File: app/src/main/java/com/bigscreen/mangindo/info/MangaInfoViewModel.java
package com.bigscreen.mangindo.info;


import com.bigscreen.mangindo.network.model.Manga;

public class MangaInfoViewModel {

    private final Manga manga;

    public MangaInfoViewModel(Manga manga) {
        this.manga = manga;
    }

    public String getAuthorName() {
        return isMangaNotNull()? manga.getAuthor() : "N/A";
    }

    public String getStatus() {
        return isMangaNotNull()? manga.getStatus() : "N/A";
    }

    public String getGenre() {
        return isMangaNotNull()? manga.getGenre() : "N/A";
    }

    public String getSummary() {
        return isMangaNotNull()? manga.getSummary() : "N/A";
    }

    private boolean isMangaNotNull() {
        return manga != null;
    }
}

File: app/src/main/java/com/bigscreen/mangindo/info/MangaInfoViewModel.kt
package com.bigscreen.mangindo.info

import com.bigscreen.mangindo.network.model.Manga


class MangaInfoViewModel(var manga: Manga?) {

    val authorName: String
        get() = manga?.author ?: "N/A"

    val status: String
        get() = manga?.status ?: "N/A"

    val genre: String
        get() = manga?.genre ?: "N/A"

    val summary: String
        get() = manga?.summary ?: "N/A"
}